Unable to provide a StackName and a ChangeSetName

When supplying a `changeSetName` 3 args are supplied forcing `suppliedStackName` is overwritten with an empty string.

https://github.com/aws-cloudformation/rain/blob/7f91d2757c1f08e7d967b8194387e3df6e225b84/internal/cmd/deploy/deploy.go#L104-L113

Here's some logs
```shell
+ rain deploy template.cfn.yaml package-repositories test --no-exec --debug --s3-bucket abcdefg-cloudformation-us-east-1 --s3-prefix 123456789012
DEBUG: Preparing template 'template.cfn.yaml'
DEBUG: Loading AWS config
DEBUG: Artifact bucket: abcdefg-cloudformation-us-east-1
DEBUG: Need another pass: 1
DEBUG: No changes this pass: 2
DEBUG: Checking current status of stack 'template-cfn'
DEBUG: Stack exists
Existing stack is empty; deleting it.
panic: unable to delete stack 'template-cfn': User: arn:aws:sts::123456789012:assumed-role/github-roles-PackageRepositories-ttcCSbjCh5Nu/9982144563.1+abd920762027cb480c2e6555743a483b159f96c3 is not authorized to perform: cloudformation:DeleteStack on resource: arn:aws:cloudformation:us-east-1:123456789012:stack/template-cfn/1de83ca0-4082-11ef-98a7-0e67c5a80ddd because no identity-based policy allows the cloudformation:DeleteStack action [recovered]
```

sorry I only got round to testing this today

--- a/internal/cmd/deploy/deploy.go
+++ b/internal/cmd/deploy/deploy.go
@@ -101,10 +101,8 @@ To list and delete changesets, use the ls and rm commands.
 
 			var suppliedStackName string
 
-			if len(args) == 2 {
+			if len(args) >= 2 {
 				suppliedStackName = args[1]
-			} else {
-				suppliedStackName = ""
 			}
 
 			// Optionally name the change set

--- a/scripts/integ.sh
+++ b/scripts/integ.sh
@@ -18,6 +18,17 @@ set -eoux pipefail
 
 ./rain rm success-test -y
 
+# Unnamed stack
+./rain deploy test/templates/success.template -y --params BucketName=ezbeardatamazon-rain-test-1
+./rain cat success
+./rain ls success
+./rain rm -y success
+
+# Change sets
+./rain deploy --no-exec test/templates/success.template success-test success-changeset-name -y --params BucketName=ezbeardatamazon-rain-test-1
+./rain ls -c success-test success-changeset-name
+./rain rm -c -y success-test success-changeset-name
+
 ./rain build AWS::S3::Bucket
 ./rain build -l
 

